### PR TITLE
opus: add support for RG tags again

### PR DESCRIPTION
**Pull request by Almoped**
Re-added Thomas's original RG code and some logic to support both R128 and RG, while preferring R128.

### CODE REVIEW

File: src/opus/opus.cc [hunk 1]
```diff
@@ -181,6 +181,15 @@ static bool update_replay_gain(OggOpusFile * opus_file,
...
     const char * track_gain = opus_tags_query(tags, "R128_TRACK_GAIN", 0);
+    bool got_r128 = true;
+
+    /* try to read RG tags if R128 didn't work*/
+    if (!album_gain && !track_gain)
+    {
+        got_r128 = false;
+        album_gain = opus_tags_query(tags, "REPLAYGAIN_ALBUM_GAIN", 0);
+        track_gain = opus_tags_query(tags, "REPLAYGAIN_TRACK_GAIN", 0);        
+    }
```
Comment (radioactiveman) [inline src/opus/opus.cc:192]: I would prefer this approach:

```c++
bool has_r128_tags = (album_gain && track_gain);

/* try to read RG tags if R128 didn't work */
if (!has_r128_tags)
{
    album_gain = opus_tags_query(tags, "REPLAYGAIN_ALBUM_GAIN", 0);
    track_gain = opus_tags_query(tags, "REPLAYGAIN_TRACK_GAIN", 0);        
}
```

File: src/opus/opus.cc [hunk 2]
```diff
@@ -192,14 +201,43 @@ static bool update_replay_gain(OggOpusFile * opus_file,
...
+            if (!track_peak)
+                track_peak = album_peak;
+
+            rg_info->album_peak = str_to_double(album_peak);
+            rg_info->track_peak = str_to_double(track_peak);
+        }
+    }
 
     AUDDBG("Album gain: %s (%f)\n", album_gain, rg_info->album_gain);
     AUDDBG("Track gain: %s (%f)\n", track_gain, rg_info->track_gain);
```
Comment (radioactiveman) [inline src/opus/opus.cc:243]: Bring back the peak log messages as well. They were removed with https://github.com/audacious-media-player/audacious-plugins/commit/78f33f7316de481d6ce34ed3614076fda700aa1c.

**Review by radioactiveman — CHANGES_REQUESTED** [review]

File: src/opus/opus.cc (hunk 1 above)
Comment (Almoped) [reply src/opus/opus.cc:192]: That makes sense. I was unsure of a clear way to have it.

**Comment by Almoped** [comment]
Trying to make friends with vscode and git it seems to be resolved now.

**Pull request merged**